operate/schema/src/test/java/io/camunda/operate/schema/IndexSchemaValidatorIT.java

```diff
@@ -0,0 +1,268 @@
+ * If the Licensee is in breach of the Conditions, this Agreement, including the rights granted under it, will automatically terminate with immediate effect.
+ *
+ * SUBJECT AS SET OUT BELOW, THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ * NOTHING IN THIS AGREEMENT EXCLUDES OR RESTRICTS A PARTY’S LIABILITY FOR (A) DEATH OR PERSONAL INJURY CAUSED BY THAT PARTY’S NEGLIGENCE, (B) FRAUD, OR (C) ANY OTHER LIABILITY TO THE EXTENT THAT IT CANNOT BE LAWFULLY EXCLUDED OR RESTRICTED.
+ */
+package io.camunda.operate.schema;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.assertj.core.api.Assertions.entry;
+
+import io.camunda.operate.JacksonConfig;
+import io.camunda.operate.connect.ElasticsearchConnector;
+import io.camunda.operate.connect.OpensearchConnector;
+import io.camunda.operate.connect.OperateDateTimeFormatter;
+import io.camunda.operate.exceptions.OperateRuntimeException;
+import io.camunda.operate.schema.IndexMapping.IndexMappingProperty;
+import io.camunda.operate.schema.elasticsearch.ElasticsearchSchemaManager;
+import io.camunda.operate.schema.indices.IndexDescriptor;
+import io.camunda.operate.schema.opensearch.OpensearchSchemaManager;
+import io.camunda.operate.schema.util.SchemaTestHelper;
+import io.camunda.operate.schema.util.TestIndex;
+import io.camunda.operate.schema.util.TestTemplate;
+import io.camunda.operate.schema.util.elasticsearch.ElasticsearchSchemaTestHelper;
+import io.camunda.operate.schema.util.opensearch.OpenSearchSchemaTestHelper;
+import io.camunda.operate.store.elasticsearch.ElasticsearchTaskStore;
+import io.camunda.operate.store.opensearch.OpensearchTaskStore;
+import java.util.Map;
+import java.util.Set;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+
+@ContextConfiguration(
+    classes = {
+      IndexSchemaValidator.class,
+      TestIndex.class,
+      TestTemplate.class,
+      ElasticsearchSchemaManager.class,
+      ElasticsearchConnector.class,
+      ElasticsearchTaskStore.class,
+      ElasticsearchSchemaTestHelper.class,
+      OpensearchSchemaManager.class,
+      OpensearchConnector.class,
+      OpensearchTaskStore.class,
+      OpenSearchSchemaTestHelper.class,
+      JacksonConfig.class,
+      OperateDateTimeFormatter.class
+    })
+public class IndexSchemaValidatorIT extends AbstractSchemaIT {
```

Comment:
My idea was to have them in the schema module so that when you develop the module further, you can build and test the module on its own and already get some test coverage. Having them in the QA module would mean you have to build most of the operate modules and potentially wait for other unrelated tests to run too.

I don't have strong feelings in either direction, it is also "just" a matter of how you want to organize the tests.

Reply:
I'm actually fine with keeping them in the schema module, I think it makes sense to have tests related to a specific module in that module rather than grouping them all under one subfolder. I just wanted to raise the question and see where thoughts were.